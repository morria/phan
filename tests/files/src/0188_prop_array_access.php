<?php
class C188 implements ArrayAccess {}
$v = new C188;
$v['key'] = 42;
class E188 {
    /** @var C188 */
    private $p;
    function f() {
        $this->p['key'] = 42;
        $this->p->key = 43;
    }
}
