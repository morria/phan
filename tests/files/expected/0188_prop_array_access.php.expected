%s:2 PhanClassContainsAbstractMethodInternal non-abstract class \C188 contains abstract internal method \ArrayAccess::offsetExists
%s:2 PhanClassContainsAbstractMethodInternal non-abstract class \C188 contains abstract internal method \ArrayAccess::offsetGet
%s:2 PhanClassContainsAbstractMethodInternal non-abstract class \C188 contains abstract internal method \ArrayAccess::offsetSet
%s:2 PhanClassContainsAbstractMethodInternal non-abstract class \C188 contains abstract internal method \ArrayAccess::offsetUnset
%s:10 PhanUndeclaredProperty Reference to undeclared property \C188->key
